### PR TITLE
[WeChat QR] Fixing issue that Hybrid Binarizer not working due to integral calculation.

### LINKED ISSUE #3156: Is there bug in hybird binarizer method in wechat qrcode?
##### System information (version)

- OpenCV => :grey_question:
- Operating System / Platform => :grey_question:
- Compiler => :grey_question:

##### Detailed description

I have tested many qrcode, there is no qrcode is decoded with hybird binarizer method, and I get the output of this method, just like this issue: https://github.com/opencv/opencv_contrib/issues/3008

### PR DESCRIPTION
**Pull request by kim-ninh**
This should fixes #3156 #3008 

The size of block array `blocks_` is `subWidth * subHeight` but the block integral size is `width * height`. I believe something wrong when calculating the `blockIntegral` with that size.

https://github.com/opencv/opencv_contrib/blob/508c8db05e438635c84825629c069252bb53771a/modules/wechat_qrcode/src/zxing/common/binarizer/hybrid_binarizer.cpp#L41

https://github.com/opencv/opencv_contrib/blob/508c8db05e438635c84825629c069252bb53771a/modules/wechat_qrcode/src/zxing/common/binarizer/hybrid_binarizer.cpp#L58

I suggest change it size to `(subWidth + 1) * (subHeight + 1)` (These extra row and column is just for padding 0 arroud). Also update the precalculation of the `blockIntegral`, calculate sum base on that `blockIntegral`

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

`
force_builders=linux,docs
`

### CODE REVIEW

**Comment by kim-ninh** [comment]
Waiting for Maintainers to trigger CI manually.

**Comment by dddzg** [comment]
There are some test cases failed in Win64,Mac and Win32.
https://pullrequest.opencv.org/buildbot/builders/precommit-contrib_windows64/builds/15419

**Comment by kim-ninh** [comment]
> There are some test cases failed in Win64,Mac and Win32. https://pullrequest.opencv.org/buildbot/builders/precommit-contrib_windows64/builds/15419

Got it. Already fix those test cases in new commit.

**Comment by dddzg** [comment]
@alalek

**Review by alalek — APPROVED** [review]
LGTM :+1:
Thank you for contribution!

**Pull request merged**
